src/main/java/com/gtnewhorizon/structurelib/item/ItemDebugStructureWriter.java

```diff
@@ -0,0 +1,187 @@
+
+            case Refresh:
+                tryMakeAndDrawBox(world);
+                break;
+            case SetCenter:
+                center = pos;
+                break;
+            case Clear:
+                StructureLib.proxy.clearHints(world);
+                corners[0] = null;
+                corners[1] = null;
+                center = null;
+                break;
+            case Build:
+                writeStructure(player);
+                break;
+        }
+    }
+
+    private void tryMakeAndDrawBox(World world) {
+        if (corners[0] != null && corners[1] != null) {
+            this.box = new Box(corners[0], corners[1]);
+
+            box.drawBoundingBox(world);
+        }
+    }
+
+    private void writeStructure(EntityPlayer player) {
+        if (corners[0] != null && corners[1] != null) {
+            ExtendedFacing facing = StructureUtility.getExtendedFacingFromLookVector(player.getLookVec());
+
+            String structureDefinition = StructureUtility.getPseudoJavaCode(player.getEntityWorld(),
+                                                                            facing,
+                                                                            box,
+                                                                            this.center != null ? this.center : Vec3Impl.NULL_VECTOR,
+                                                                            player.isSneaking());
+
+            StructureLib.LOGGER.info(structureDefinition);
+        }
+    }
+
+    @SideOnly(Side.CLIENT)
+    @Override
+    public IIcon getIconFromDamage(int damage)
+    {
+        return damage == Usage.Clear.ordinal() ? this.eraser : super.getIconFromDamage(damage);
+    }
+
+    @SideOnly(Side.CLIENT)
+    @Override
+    public void registerIcons(IIconRegister iconRegister)
+    {
+        super.registerIcons(iconRegister);
+        this.eraser = iconRegister.registerIcon(MOD_ID + ":itemDebugStructureEraser");
+    }
+
+    @Override
+    public String getItemStackDisplayName(ItemStack itemStack) {
+        return super.getItemStackDisplayName(itemStack) + " (" + StatCollector.translateToLocal("item.structurelib.debugStructureWriter.mode." + this.mode.ordinal()) + ")";
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public void addInformation(ItemStack itemStack, EntityPlayer player, List description, boolean p_77624_4_) {
+        String mode = EnumChatFormatting.DARK_AQUA +
+                      StatCollector.translateToLocal("item.structurelib.debugStructureWriter.desc.0") +
+                      " (" + StatCollector.translateToLocal("item.structurelib.debugStructureWriter.mode." + this.mode.ordinal()) + ")";
+        description.add(mode);
+
+        switch (this.mode) {
+            case SetCorners:
+                description.add(StatCollector.translateToLocal("item.structurelib.debugStructureWriter.desc.1"));
```

Comment:
Okay so, hear me out. The Collection interface has this method called addAll().
Or in general a more parametric approach would be poggers. 
Another solution is instead of using a switch, have the descriptions lang strings stored in the mode enum values.

Here, it's fine to keep it like it is. But in the future, might be smart to switch it out for something less pepega. All Clear.